### PR TITLE
[Embed Override] Validate embed override hidden_dim at ingestion time

### PR DESCRIPTION
**Pull request by fortunecookiee**
## Motivation

`positional_embed_overrides` are eventually scattered into the input embedding buffer in `ModelRunner`:

```python
kwargs["input_embeds"][forward_batch.replace_positions] = (
    forward_batch.replace_embeds.to(kwargs["input_embeds"].dtype)
)
```

If a caller supplies an override tensor whose `hidden_dim` does not match the model's `hidden_size`, nothing checks it on the way in. The mismatch only fails at that index-assign — i.e. inside the **scheduler process**, after tokenization, batching and scheduling have already happened. The result is a bare shape `RuntimeError` that is hard to attribute back to the offending request, and that surfaces to the caller as an opaque internal error rather than an input-validation error. It is also easy to trigger with a simple client mistake (e.g. sending embeddings from a differently-sized model).

## Modifications

- Add `PositionalEmbeds.validate_hidden_dim(expected_hidden_dim)`, which raises a `ValueError` naming both the actual and the expected size.
- Add `TokenizerManager._validate_positional_embed_overrides_hidden_dim()`, which handles both shapes that reach the scheduler — the single-request `PositionalEmbeds` and the score-request `List[Optional[PositionalEmbeds]]` — and skips `None` entries.
- Call it from `_tokenize_one_request` on both the `GenerateReqInput` and `EmbeddingReqInput` branches. On the embedding branch the check runs *after* placeholder resolution, so overrides supplied via `embed_override_token_id` / `embed_overrides` are covered too.

Validation is against `model_config.hidden_size` (not `spec_hidden_size`), which is the dim of the buffer being scattered into. It is a no-op for requests that carry no overrides, so there is no cost on the normal path.

## Accuracy Tests

No behavior change for valid requests — this only converts an already-failing case into a clearer, earlier failure. Added unit tests to `test/registered/unit/managers/test_embed_overrides.py` (registered CPU suite) covering: matching dim accepted, mismatched dim rejected with both sizes in the message, `None` is a no-op, list form skips `None` entries, and list form rejects a mismatched member.

## Benchmarking and Profiling

Not applicable — validation runs only when overrides are present and is a single shape comparison per request.

## Checklist

- [x] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#code-formatting-with-pre-commit).
- [x] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#running-unit-tests-adding-to-ci).
- [x] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/developer_guide/contribution_guide.html#writing-documentation-running-docs-ci).
- [x] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/developer_guide/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/developer_guide/accuracy_evaluation.html).
- [x] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [x] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

---
### CI States

Latest PR Test (Base): :x: [Run #34899609511](https://github.com/sgl-project/sglang/actions/runs/34899609511)
Latest PR Test (Extra): :x: [Run #34899609492](https://github.com/sgl-project/sglang/actions/runs/34899609492)
Latest PR Test (AMD ROCm 10): :x: [Run #34899609528](https://github.com/sgl-project/sglang/actions/runs/34899609528)

### CODE REVIEW

**Comment by fortunecookiee** [comment]
Rebased onto current `main` and hardened after a self-review found the validator left the worst failure mode open.

`positional_embed_overrides` is typed `Any` and the request dataclasses are bound directly as FastAPI bodies, so arbitrary JSON reached the validator. `{"positional_embed_overrides": [null]}` passed validation, encoded fine (msgpack encoding is untyped), then failed `msgpack_decode` inside the scheduler's receive loop — which only catches `zmq.ZMQError` — taking the scheduler process down. A dict was iterated by key and raised `AttributeError` (500), and a scalar raised `TypeError` (500).

Changes:
- The validator now rejects non-`PositionalEmbeds` values with `ValueError` (→ HTTP 400) instead of forwarding them.
- `PositionalEmbeds.__post_init__` enforces a 2-D normalized tensor. `validate_hidden_dim` only inspects the last dim, so `[N, 1, hidden_dim]` satisfied both it and the positions-length check, then failed in `ScheduleBatch.prepare_for_extend`'s `torch.cat` and aborted every co-batched request.
- An empty `embeds` list now raises `ValueError` rather than letting `torch.cat([])` raise `RuntimeError`, which the entrypoints don't map and surfaced as a 500.
